src/client/public/scripts/components/nav-element.ts

```diff
@@ -6,20 +6,29 @@ import {BaseElement, property} from './base-element.js';
 
 export class NavElement extends BaseElement {
   @property() data: api.UserResponse|null = null;
+  @property() documentUrl = document.location.pathname;
 
   async connectedCallback() {
     const response = await fetch('/api/user', {credentials: 'include'});
     // TODO: change this to use the JSON API response
     this.data = (await response.json()).data as api.UserResponse;
+
+    document.body.addEventListener('url-changed', this.urlChanged.bind(this));
```

Comment:
I feel like you could just inline this method as a fat arrow function.

Reply:
Done.